src/tests/createHeadlessForm.test.js

```diff
@@ -776,6 +780,74 @@ describe('createHeadlessForm', () => {
       });
     });
 
+    it('support "radio" field boolean type', () => {
+      const { fields, handleValidation } = createHeadlessForm(schemaInputTypeRadioBoolean);
+
+      expect(fields).toMatchObject([
+        {
+          description: 'Are you over 18 years old?',
+          label: 'Over 18',
+          name: 'over_18',
+          options: [
+            {
+              label: 'Yes',
+              value: true,
+            },
+            {
+              label: 'No',
+              value: false,
+            },
+          ],
+          required: true,
+          schema: expect.any(Object),
+          type: 'radio',
+        },
+      ]);
+
+      assertOptionsAllowed({
+        handleValidation,
+        fieldName: 'over_18',
+        validOptions: [true, false],
```

Comment:
bug: Something is still wrong in this code because if I pass `"true"` or `"false"`, it's still valid when it should not be.

For example:

```js
  const validateForm = (vals) => friendlyError(handleValidation(vals));

  expect(validateForm({over_18: 'true'})).toEqual({
    over_18: 'The option "true" is not valid.',
  });
```

Reply:
Looks like yup casts 'true' to `true` when using boolean()

Reply:
`
      validateForm({ over_18: 'true' });
      validateForm({ over_18: 'false' });
      expect(validateForm({over_18: 'maybe'})).toEqual({
        over_18: 'The option "maybe" is not valid.',
      });
`

![Screenshot 2024-08-19 at 21 19 02](https://github.com/user-attachments/assets/aa1bacfd-d712-4b71-b8fa-5acad25d700f)
![Screenshot 2024-08-19 at 21 19 08](https://github.com/user-attachments/assets/a2ed5c2e-551e-4cce-88bf-2f2f9d889b68)

Reply:
Saw the latest commit, looks good! Release a new dev version with it, and once our internal MR has green tests, then we can merge this and release the final version.

Reply:
@Arcombe-Remote , I'm optimistic the internal MR will pass, so feel free to rever the version back to what it is in main (package and package-lock) and I'll aprove this right away :)

Reply:
@brennj could you re-approved this MR? It was already reviewed/approved by me, but I had to push a commit to revert to the main version. That way I can unblock @Arcombe-Remote when he starts his day tomorrow.

Reply:
course